ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/BlockHeader.java

```diff
@@ -217,7 +218,7 @@ public void writeTo(final RLPOutput out) {
     out.writeBytes(mixHashOrRandom);
     out.writeLong(nonce);
     if (baseFee != null) {
-      out.writeLongScalar(baseFee);
+      out.writeUInt256Scalar(baseFee);
```

Comment:
there appears to be a difference in writeUInt256Scalar and writeLongScalar.   writeLongScalar is always going to encode 8 bytes, whereas UInt256Scalar appears to trim leading zeros...  

This could be an encoding difference that might lead to a problem.

Reply:
I double checked and the output is the same, indeed also the javadoc says that both are encoded without leading zero:
[writeLongScalar](https://github.com/fab-10/besu/blob/b8816e6210c33b869146108b8a35814f55344141/ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/RLPOutput.java#L104)
[writeUInt256Scalar](https://github.com/fab-10/besu/blob/b8816e6210c33b869146108b8a35814f55344141/ethereum/rlp/src/main/java/org/hyperledger/besu/ethereum/rlp/RLPOutput.java#L76)

I have add a[ unit test](https://github.com/fab-10/besu/blob/b8816e6210c33b869146108b8a35814f55344141/ethereum/rlp/src/test/java/org/hyperledger/besu/ethereum/rlp/BytesValueRLPOutputTest.java#L181) to verify that the encoding is the same.

Am I still missing something?

Reply:
You are right - both trim leading zeros.  I would swear the writeLongScalar implementation I looked at said it would write exactly 8 bytes.  Looking on main, I see that writeLongScalar is explicitly `writeBytes(Bytes.minimalBytes(v))` 🤷